### PR TITLE
Include metadata in `AssetSerializer`

### LINKED ISSUE #78: API /assets/: add an option to include metadata into the listing
For many use cases some metadata per each item out be desired in `dandi-cli`:
- for `ls` (not implemented yet): to list assets on remote end as it does locally
- for `download` (WiP): to obtain `uploaded_mtime` and `uploaded_nwb_object_id` (and possibly an extended list of digests) to make decision on either to download a newer version of the asset
- for `upload` (there is no API yet so we talk to girder directly ATM, but in the future): similar to `download` to make a decision either re-upload of the asset is needed

With current API it would mean an additional query per each asset to `/asset/{uuid}` end-point increasing the burden on the server and taking it longer overall. Adding some option like `include_metadata=true` into `/assets/` query would allow to avoid this additional interaction at a much smaller (IMHO) cost -- AFAIK metadata records are quite lean ATM and ok to be transferred... well, we could further improve it to make it actually into a `include_metadata=regex` instead of just a "bool".  That would allow clients to pick metadata to be transferred, thus decreasing the transfer burden etc (e.g. for upload/download could be `include_metadata=(uploaded_.*|md5)` or alike).

### PR DESCRIPTION
**Pull request by AlmightyYakob**
Fixes #78

### CODE REVIEW

File: publish/views/asset.py [hunk 1]
```diff
@@ -46,7 +42,7 @@ class AssetViewSet(NestedViewSetMixin, DetailSerializerMixin, ReadOnlyModelViewS
 
     permission_classes = [IsAuthenticatedOrReadOnly]
     serializer_class = AssetSerializer
-    serializer_detail_class = AssetDetailSerializer
+    serializer_detail_class = AssetSerializer
```
Comment (brianhelba) [inline publish/views/asset.py:45]: No need to include `serializer_detail_class` if it's the same as `serializer_class`.

Comment (brianhelba) [reply publish/views/asset.py:45]: Additionally, we can remove the lines with `queryset_detail` and the `DetailSerializerMixin` itself, if we're not going to use it for now.

Comment (AlmightyYakob) [reply publish/views/asset.py:45]: > No need to include `serializer_detail_class` if it's the same as `serializer_class`.
> 
> [chibisov.github.io/drf-extensions/docs/#detailserializermixin](https://chibisov.github.io/drf-extensions/docs/#detailserializermixin)
> 
> > If you use DetailSerializerMixin and don't specify serializer_detail_class attribute, then serializer_class will be used.

If I make the change you suggested, I get the following error:
```
'AssetViewSet' should include a 'serializer_detail_class' attribute
```

Do you know a way around this?

Comment (AlmightyYakob) [reply publish/views/asset.py:45]: > Additionally, we can remove the lines with `queryset_detail` and the `DetailSerializerMixin` itself, if we're not going to use it for now.

Actually, if I make these changes as well, the errors go away. If we're not going to use the detail, I'm fine with committing that.

**Pull request merged**